packages/flutter_tools/test/general.shard/custom_devices/custom_device_test.dart

```diff
@@ -519,6 +520,63 @@ void main() {
       ProcessManager: () => FakeProcessManager.any()
     }
   );
+
+  testWithoutContext('CustomDevice screenshotting', () async {
+    bool screenshotCommandWasExecuted = false;
+
+    final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: testConfig.screenshotCommand,
+        onRun: () => screenshotCommandWasExecuted = true,
+      )
+    ]);
+
+    final MemoryFileSystem fs = MemoryFileSystem.test();
+    final File screenshotFile = fs.file('screenshot.png');
+
+    final CustomDevice device = CustomDevice(
+      config: testConfig,
+      logger: BufferLogger.test(),
+      processManager: processManager
+    );
+
+    expect(device.supportsScreenshot, true);
+
+    await device.takeScreenshot(screenshotFile);
+    expect(screenshotCommandWasExecuted, true);
+    expect(screenshotFile.existsSync(), true);
```

Comment:
nit: you can use `exists` from package:file_testing

Reply:
done